src/java/org/apache/cassandra/db/rows/AbstractCell.java

```diff
@@ -114,14 +119,44 @@ public Cell<?> clone(ByteBufferCloner cloner)
     // note: while the cell returned may be different, the value is the same, so if the value is offheap it must be referenced inside a guarded context (or copied)
     public Cell<?> updateAllTimestamp(long newTimestamp)
     {
-        return new BufferCell(column, isTombstone() ? newTimestamp - 1 : newTimestamp, ttl(), localDeletionTime(), buffer(), path());
+        CellPath newPath = null;
+        if (path() != null) {
+            byte[] newUUID = nextTimeUUIDAsBytes(newTimestamp);
+            newPath = CellPath.create(ByteBuffer.wrap(newUUID));
+            logger.debug("timestamp: {}    newPath: {}", newTimestamp, newPath.get(0));
+        }
+        return new BufferCell(column, isTombstone() ? newTimestamp - 1 : newTimestamp, ttl(), localDeletionTime(), buffer(), newPath);
```

Comment:
Would this not change all cell paths, including the ones used in maps, breaking them?

Shouldn't this be done by `ListType` rather than the generic `Cell` implementation?